README.md

```diff
@@ -0,0 +1,38 @@
+# eslint-config-lintification
+
+A suite of battle-tested linting rules for any modern Node.js application, with full support for the most common TypeScript rules.
+
+
+
+## Table of Contents
+- [eslint-config-lintification](#eslint-config-lintification)
+  - [Table of Contents](#table-of-contents)
+  - [Usage](#usage)
+    - [Install](#install)
+    - [Extend the Configuration](#extend-the-configuration)
+  - [License](#license)
+
+
+
+## Usage
+
+### Install
+```bash
+$ npm install -D eslint-config-lintification eslint @stylistic/eslint-plugin@2.10.1 @typescript-eslint/eslint-plugin@6.7.4 eslint-plugin-file-progress@1.3.0 eslint-plugin-import@2.28.1 eslint-plugin-import-newlines@1.3.4 eslint-plugin-jest@27.4.2 eslint-plugin-newline-destructuring@1.2.2 eslint-plugin-simple-import-sort@10.0.0 eslint-import-resolver-typescript@3.6.1
+```
+
+### Extend the Configuration
+
+Either add `"extends": "lintification"` to your linting configuration or add the below to your package.json to use the rules with no modifications:
+```json
+{
+    "eslintConfig": {
+        "extends": "lintification"
+    }
+}
+```
```

Comment:
Can you reorganize the sections to be the following?
- install and add to package.json (should just be `npm install eslint-config-linitification` and then add to package.json)
- How to extend/overwrite rules with an example for how to add/disable/change rules
- How to contribute to the package (shouldn't need to install all of those manually, they should be able to be marked as dev dependencies in the package.json)

Reply:
> How to install and add to package.json (should just be npm install eslint-config-lintification and then add eslintConfig to package.json)

This isn't necessarily true. Peer dependencies need to be installed as well in order for things to work (at least when testing locally with `npm link`). `npm link` has led to differences with linting rules packages like this before though, so I'd be willing to go back and edit that once it's ready for consumption for the first time if the peers are only necessary for local linking.

Reply:
Could you also propose what you're wanting out of this documentation? I'm unclear on what you actually want changed. Using [airbnb-base's README](https://www.npmjs.com/package/eslint-config-airbnb-base) as an example, I opted not to include instructions on how to override rules, since that's something that can easily be searched outside the confines of this specific linting config in eslint's own documentation.

Reply:
@beanc16 Peer dependencies don't need to be installed manually, they should be automatically installed when you install the package as well.

For the readme, something like this:
````suggestion
### Installation
`npm install eslint-config-lintification`

Then, add the configuration to your `package.json`:
```json
{
    "eslintConfig": {
        "extends": "eslint-config-lintification"
    }
}
```

Finally, ensure you have a linting script added to your `package.json`, like so:
```json
{
    "scripts": {
        "lint": "eslint --cache --ext js --ext ts ./",
        "lint:fix": "eslint --cache --fix --ext js --ext ts ./"
    }
}
```

Then, just run `npm run lint` to run the linter, and run `npm run lint:fix` to fix any issues that can automatically be fixed by the linter.

### Extend the Configuration
To add new or overwrite existing rules by including them in the `rules` property of the `eslintConfig` object in your `package.json`.
```json
{
    "eslintConfig": {
        "extends": "eslint-config-lintification",
        "rules": {
            "object-curly-spacing": "off"
        }
    }
}
```
````

Not worried about the contributing section yet, we can add that later.

Reply:
> Peer dependencies don't need to be installed manually, they should be automatically installed when you install the package.

Must've been an issue with local `npm link`ing that didn't install those for me then. We should figure out the simplest way to link locally for maintenance and contribution documentation in the future.

One thing I've found is a `--resolve-plugins-relative-to` flag that'd be used similar to this:
```
        "lint": "eslint --resolve-plugins-relative-to './path-on-your-local-machine-to/eslint-config-lintification' --cache --ext js --ext ts ./",
```

Though that's also imperfect and throws this error:
```
Error: Error while loading rule 'jest/no-deprecated-functions': Unable to detect Jest version - please ensure jest package is installed, or otherwise set version explicitly
```

From my research, seems like what I encountered is just an issue with `npm link` and peer dependencies.

Reply:
It's no better nor worse, the `jest/no-deprecated-functions` error still occurs.

Reply:
I applied the changes to the README with some notable changes:
1. Added `-D` to the suggestion so that it'd be installed as a dev dependency in consuming projects.
2. All `extends` fields should evaluate to `lintification`, not `eslint-config-lintification`. You'll notice we don't do that for any other config we extend either, as that's how eslint processes it. https://github.com/nick-w-nick/eslint-config-lintification/blob/c459ec9f0fd1a30012dc389b579c5b808dab2f68/eslint.config.js#L8

Reply:
@beanc16 How does that work for when the package gets consumed though since the dev dependencies will be removed?